### PR TITLE
Replace Vue Router to InertiaJS, Fixing typo in getting buttonKey

### PR DESCRIPTION
**Pull request by hsuan1117**

### CODE REVIEW

File: resources/field.js [hunk 1]
```diff
@@ -10,7 +11,7 @@ export default {
     reload() {
       if (this.field.reload && queue.allowsReload()) {
         window.setTimeout(() => {
-          this.$router.go();
+          Inertia.reload()
```
Comment (sietse85) [inline resources/field.js:14]: This change does not seem to have any effect? How did you test this?

File: package.json [hunk 2]
```diff
@@ -29,6 +29,7 @@
         "vue-loader": "^16.8.3"
     },
     "dependencies": {
+        "@inertiajs/inertia": "^0.11.0",
```
Comment (sietse85) [inline package.json]: I think Nova already includes inertia and this addition is not neccesary

**Review by sietse85 — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/ButtonController.php [hunk 3]
```diff
@@ -22,7 +22,7 @@ public function handle(NovaRequest $request)
         $event = $request->get('event');
         $resource = $request->findModelQuery()->firstOrFail();
 
-        event(new $event($resource, $request->get('buttonKey')));
+        event(new $event($resource, $request->route('buttonKey')));
```
Comment (sietse85) [inline src/Http/Controllers/ButtonController.php:25]: this is actually not a typo, this does what it needs to do get the buttonKey attribute from the route.

**Review by sietse85 — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/ButtonController.php (hunk 3 above)
Comment (hsuan1117) [reply src/Http/Controllers/ButtonController.php:25]: I just test it and I get `null` from `$request->get('buttonKey')`

File: resources/field.js (hunk 1 above)
Comment (hsuan1117) [reply resources/field.js:14]: You can test it by using `->reload()` function.
It should throw errors about `Cannot read properties of undefined (reading go)`, and it works after change to Inertia reload function

**Comment by sietse85** [comment]
@hsuan1117 thank you, changed in v1.0.15

**Pull request merged**